subsys/logging/log_core.c

```diff
@@ -396,12 +396,24 @@ void log_generic(struct log_msg_ids src_level, const char *fmt, va_list ap)
 	} else {
 		log_arg_t args[LOG_MAX_NARGS];
 		u32_t nargs = count_args(fmt);
+		u32_t mask;
 
 		__ASSERT_NO_MSG(nargs < LOG_MAX_NARGS);
 		for (int i = 0; i < nargs; i++) {
 			args[i] = va_arg(ap, log_arg_t);
 		}
 
+		if (enable_strdup && ( mask = z_log_get_s_mask(fmt, nargs ))) {
+			while (mask) {
+				u32_t idx =31 - __builtin_clz(mask);
+				const char *str = (const char *)args[idx];
+				if (!log_is_strdup(str) &&
```

Comment:
need empty line after declaration.

Reply:
any reason for not adding `!is_rodata(str)` to if conditions like in `detect_missed_strdup`?

Reply:
The reason is following: is_rodata(str) is checked inside log_strdup, so it would be checked twice (unless compiler optimize it)